oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -0,0 +1,784 @@
+        catch (StorageException e) {
+            LOG.info("Error adding metadata record. metadataName={} length={}", name, recordLength, e);
+            throw new DataStoreException(e);
+        }
+        catch (URISyntaxException |  IOException e) {
+            throw new DataStoreException(e);
+        }
+    }
+
+    @Override
+    public DataRecord getMetadataRecord(String name) {
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        long start = System.currentTimeMillis();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            CloudBlobDirectory metaDir = getAzureContainer().getDirectoryReference(META_DIR_NAME);
+            CloudBlockBlob blob = metaDir.getBlockBlobReference(name);
+            if (!blob.exists()) {
+                // TODO: Is returning null the right thing to do? FileStores think so.
+                LOG.warn("Trying to read missing metadata. metadataName={}", name);
+                return null;
+            }
+            blob.downloadAttributes();
+            long lastModified = blob.getProperties().getLastModified().getTime();
+            long length = blob.getProperties().getLength();
+            AzureBlobStoreDataRecord record =  new AzureBlobStoreDataRecord(this,
+                                                connectionString,
+                                                containerName, new DataIdentifier(name),
+                                                lastModified,
+                                                length,
+                                                true);
+            LOG.debug("Metadata record read. metadataName={} duration={} record={}", name, (System.currentTimeMillis() - start), record);
+            return record;
+
+        } catch (StorageException e) {
+            LOG.info("Error reading metadata record. metadataName={}", name, e);
+            throw new RuntimeException(e);
+        } catch (Exception e) {
+            LOG.debug("Error reading metadata record. metadataName={}", name, e);
+            throw new RuntimeException(e);
+        } finally {
+            if (null != contextClassLoader) {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
+        }
+    }
+
+    @Override
+    public List<DataRecord> getAllMetadataRecords(String prefix) {
+        if (null == prefix) {
+            throw new NullPointerException("prefix");
+        }
+        long start = System.currentTimeMillis();
+        final List<DataRecord> records = Lists.newArrayList();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            CloudBlobDirectory metaDir = getAzureContainer().getDirectoryReference(META_DIR_NAME);
+            for (ListBlobItem item : metaDir.listBlobs(prefix)) {
```

Comment:
What is the default for number of items returned in 1 call? In S3 its 1000 so it is safe to just have 1 call but if the default if low then we would need to use RecordsIterator to create the list.